src/kronecker.jl

```diff
@@ -0,0 +1,285 @@
+
+"""
+    kron(A::LinearMap, B::LinearMap)
+
+Construct a `KroneckerMap <: LinearMap` object, a (lazy) representation of the
+Kronecker product of two `LinearMap`s. One of the two factors can be an `AbstractMatrix`,
+which is then promoted to a `LinearMap` automatically. To avoid fallback to
+the generic [`Base.kron`](@ref), there must be a `LinearMap` object among the
+first 8 arguments in usage like `kron(A, B, Cs...)`. For convenience, one can
+also use `A ⊗ B` or `⊗(A, B, Cs...)` (typed as `\\otimes+TAB`) to construct the
+`KroneckerMap`.
+
+If `A`, `B`, `C` and `D` are linear maps of such size that one can form the matrix
+products `A*C` and `B*D`, then the mixed-product property `(A⊗B)*(C⊗D) = (A*C)⊗(B*D)`
+holds. Upon vector multiplication, this rule is checked for applicability.
+
+# Examples
+```jldoctest; setup=(using LinearAlgebra, SparseArrays, LinearMaps)
+julia> J = LinearMap(I, 2) # 2×2 identity map
+LinearMaps.UniformScalingMap{Bool}(true, 2)
+
+julia> E = spdiagm(-1 => trues(1)); D = E + E' - 2I;
+
+julia> Δ = kron(D, J) + kron(J, D); # discrete 2D-Laplace operator
+
+julia> Matrix(Δ)
+4×4 Array{Int64,2}:
+ -4   1   1   0
+  1  -4   0   1
+  1   0  -4   1
+  0   1   1  -4
+```
+"""
+Base.kron(A::LinearMap, B::LinearMap) = KroneckerMap{promote_type(eltype(A), eltype(B))}((A, B))
+Base.kron(A::LinearMap, B::KroneckerMap) = KroneckerMap{promote_type(eltype(A), eltype(B))}(tuple(A, B.maps...))
+Base.kron(A::KroneckerMap, B::LinearMap) = KroneckerMap{promote_type(eltype(A), eltype(B))}(tuple(A.maps..., B))
+Base.kron(A::KroneckerMap, B::KroneckerMap) = KroneckerMap{promote_type(eltype(A), eltype(B))}(tuple(A.maps..., B.maps...))
+Base.kron(A::LinearMap, B::LinearMap, Cs::LinearMap...) = KroneckerMap{promote_type(eltype(A), eltype(B), map(eltype, Cs)...)}(tuple(A, B, Cs...))
+Base.kron(A::AbstractMatrix, B::LinearMap) = kron(LinearMap(A), B)
+Base.kron(A::LinearMap, B::AbstractMatrix) = kron(A, LinearMap(B))
+# promote AbstractMatrix arguments to LinearMaps, then take LinearMap-Kronecker product
+for k in 3:8 # is 8 sufficient?
+    Is = ntuple(n->:($(Symbol(:A,n))::AbstractMatrix), Val(k-1))
+    # yields (:A1, :A2, :A3, ..., :A(k-1))
+    L = :($(Symbol(:A,k))::LinearMap)
+    # yields :Ak
+    mapargs = ntuple(n -> :(LinearMap($(Symbol(:A,n)))), Val(k-1))
+    # yields (:LinearMap(A1), :LinearMap(A2), ..., :LinearMap(A(k-1)))
+
+    @eval Base.kron($(Is...), $L, As::Union{LinearMap,AbstractMatrix}...) =
+        kron($(mapargs...), $(Symbol(:A,k)), convert_to_lmaps(As...)...)
+end
+
+struct KronPower{p}
+    function KronPower(p::Integer)
+        p > 1 || throw(ArgumentError("the Kronecker power is only defined for exponents larger than 1, got $k"))
+        return new{p}()
+    end
+end
+
+"""
+    ⊗(k::Integer)
+
+Construct a lazy representation of the `k`-th Kronecker power `A^⊗(k) = A ⊗ A ⊗ ... ⊗ A`,
+where `A` can be an `AbstractMatrix` or a `LinearMap`.
+"""
+⊗(k::Integer) = KronPower(k)
+
+⊗(a, b, c...) = kron(a, b, c...)
```

Comment:
Cool, that's because we "own" `⊗`, right? I guess we can do the same with
```julia
Base.:(^)(A::Union{LinearMap,AbstractMatrix}, ::KronPower{p})
```